staging/src/k8s.io/component-base/cli/flag/string_ref_flag.go

```diff
@@ -0,0 +1,54 @@
+/*
+Copyright 2014 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flag
+
+import (
+	"fmt"
+
+	"github.com/spf13/pflag"
+)
+
+// StringRefFlag is a string flag compatible with pflag.Value
+type StringRefFlag struct {
+	value *string
+}
+
+var _ pflag.Value = &StringRefFlag{}
+
+// NewStringRef returns a string value which stores into the provided string pointer.
+func NewStringRef(value *string) *StringRefFlag {
+	return &StringRefFlag{value: value}
+}
+
+func (f *StringRefFlag) String() string {
+	if f.value == nil {
+		return "<nil>"
+	}
+	return *f.value
+}
+
+func (f *StringRefFlag) Set(value string) error {
+	if f.value == nil {
+		return fmt.Errorf("no target (nil pointer to string)")
+	}
```

Comment:
it's a programming error to initialize a StringRefFlag with a nil pointer it can never store a value into... not sure we need to handle that case (or should handle it in the constructor by panicking or something)

Reply:
This comment is irrelevant after we drop our custom string ref impl in favor of wrapping the default cobra one, see https://github.com/kubernetes/kubernetes/pull/130916#discussion_r2003390801